frontend/src/main/components/Users/UsersTable.js

```diff
@@ -1,33 +1,116 @@
 import React from "react";
 import OurTable from "main/components/OurTable"
-
-const columns = [
-    {
-        Header: 'id',
-        accessor: 'id', // accessor is the "key" in the data
-    },
-    {
-        Header: 'First Name',
-        accessor: 'givenName',
-    },
-    {
-        Header: 'Last Name',
-        accessor: 'familyName',
-    },
-    {
-        Header: 'Email',
-        accessor: 'email',
-    },
-    {
-        Header: 'Admin',
-        id: 'admin',
-        accessor: (row, _rowIndex) => String(row.admin) // hack needed for boolean values to show up
-    },
-];
+import { Button } from "react-bootstrap";
+import { useBackendMutation } from "main/utils/useBackend";
+import { cellToAxiosParamsDelete, cellToAxiosParamsUnhide, onDeleteSuccess } from "main/utils/usersUtils"
 
 export default function UsersTable({ users }) {
+
+    // Stryker disable all : hard to test for query caching
+    const deleteMutation = useBackendMutation(
+        cellToAxiosParamsDelete,
+        { onSuccess: onDeleteSuccess },
+        ["/api/admin/users"]
+    );
+
+    const unhideMutation = useBackendMutation(
+        cellToAxiosParamsUnhide,
+        { onSuccess: onDeleteSuccess },
+        ["/api/admin/users"]
+    );
+    // Stryker restore all
+
+    const deleteCallback = async (cell) => {
+        deleteMutation.mutate(cell);
+    }
+
+    const unhideCallback = async (cell) => {
+        unhideMutation.mutate(cell);
+    }
+
+    const columns = [
+        {
+            Header: 'id',
+            accessor: 'id', // accessor is the "key" in the data
+        },
+        {
+            Header: 'First Name',
+            accessor: 'givenName',
+        },
+        {
+            Header: 'Last Name',
+            accessor: 'familyName',
+        },
+        {
+            Header: 'Email',
+            accessor: 'email',
+        },
+        {
+            Header: 'Admin',
+            id: 'admin',
+            accessor: (row, _rowIndex) => String(row.admin) // hack needed for boolean values to show up
+        },
+        {
+            Header: 'Hidden',
+            id: 'hidden',
+            accessor: (row, _rowIndex) => String(row.hidden) // hack needed for boolean values to show up
+        }
+    ];
+
+    /*
+    const columnsWithHide = [
+        ...columns,
+        ButtonColumn("Hide", "danger", deleteCallback, "UsersTable"),
+        ButtonColumn("Unhide", "success", unhideCallback, "UsersTable")
+    ];
```

Comment:
Why is this commented out code here?

Reply:
This is a leftover code snippet. Initially I want to create hide/unhide button for every user and they are all enabled. Later I wrote the version that will disable buttons based on the "hidden" and "admin"attribute. I will remove this.